### ISSUE TITLE
Feature request: match concatenation of URL and title

### ISSUE THREAD

**Issue by infokiller** [fwextensions/QuicKey#30]
Thanks for this extension, so far it's been much better than others I tried.

From my testing, it seems that the query is matched against either the URL or the title of the tab, but not both.
For example, the current tab with this issue a title of the form 'New Issue ... Quick Tabs', while the URL is `github.com/.../QuickKey`. However, If I use the query "qkgh", it doesn't match the tab. "ghqk" does match because it matches the URL, but I would expect the former to also match.

Is it possible to support this?

Thanks!

**Comment by infokiller** [comment]
Sorry, just noticed this is a dup of https://github.com/fwextensions/QuicKey/issues/11
Closing.

**Comment by fwextensions** [comment]
Thanks for trying to keep the issues list clean, but this isn't quite a duplicate of #11, which is about matching space-delimited strings against the title+URL in any order.  Since you're typing letters with no spaces, matching multiple individual tokens isn't an issue, and could be handled by just scoring the query against a `title + url` string.  Highlighting the matching letters would require some extra handling, since the title and URL are rendered separately.  Or I could just not highlight anything if the only match is the concatenated string.  

Reopening this for now.

**Comment by infokiller** [comment]
Thanks for clarifying and reopening!
Could you briefly describe the current logic with regards to whether the title or URL is used for matching?
Do you try to match the title, and only if that fails you match against URLs?

**Comment by fwextensions** [comment]
No, both the title and URL are scored, and whichever score is higher is used as the tab's score when sorting the list of results.  So adding a third score for the `title + url` string would be straightforward.

**Comment by infokiller** [comment]
Note that it's actually two added scores: `title+url` and `url+title`. Otherwise in my example above only one of `qkgh` or `ghqk` will match.

**Comment by fwextensions** [comment]
The doubles the number of strings it has to match, which will slow it down a bit, though probably not enough to be noticeable.
